tss-esapi/src/abstraction/nv.rs

```diff
@@ -17,27 +20,20 @@ pub fn read_full(
     auth_handle: NvAuth,
     nv_index_handle: NvIndexTpmHandle,
 ) -> Result<Vec<u8>> {
-    let maxsize = context
-        .get_tpm_property(PropertyTag::NvBufferMax)?
-        .unwrap_or(512) as usize;
-
-    let nv_idx = TpmHandle::NvIndex(nv_index_handle);
-    let nv_idx = context.execute_without_session(|ctx| ctx.tr_from_tpm_public(nv_idx))?;
-    let nv_idx: NvIndexHandle = nv_idx.into();
-
-    let (nvpub, _) = context.execute_without_session(|ctx| ctx.nv_read_public(nv_idx))?;
-    let nvsize = nvpub.data_size();
-
+    let mut rw = NvOpenOptions::new().open(context, auth_handle, nv_index_handle)?;
     let mut result = Vec::new();
-    result.reserve_exact(nvsize);
-
-    for offset in (0..nvsize).step_by(maxsize) {
-        let size: u16 = std::cmp::min(maxsize, nvsize - offset) as u16;
+    result.reserve_exact(rw.size());
```

Comment:
I know this is left from the previous iteration of `read_full` - you could compact this into [`Vec::with_capacity(rw.size())`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.with_capacity)